Styles branch

Manuelhh opened the pull request:
1 
Created styles branch
Deleted Header component file and its styles sass doc.
Deleted Header directory from src folder.
Deleted button & file input from app component
Deleted Header component request from App doc
Deleted styles rules styling header component from styles.scss doc and stablished styles.scss doc as a global styles doc. 
Added App.scss doc and stablished/requested it in App.tsx for styling. 
Imported SASS variables to App.scss
Deleted styles.scss import from App.tsx as it was requested and executed already in index.tsx
Removed empty code lines from variables.scss
Now clean canvas to work on styles 


2
Created main app container in app.tsx
Started styling main app container; 
centered it, added a border with box shadow and added margins.

3 
Styled body scroll bar

4
Min changes in styles

5
Added 5 divs to App components. Each represents a component that will replace div shortly. 

6
Styled temporary divs in App components representing the spaces the components will take.
Added responsiveness at 768px. For app component expanding it from 65vw to 90vw.  

7
Added 1rem padding to body.
Created components directory inside src folder.
Created Header directory to store Header component.
Inside Header directory created Header component tsx file using arrow function created Header component. Creadete Header.scss file for styling Header component.
Imported ok200 image from media directory  as logo inside Header.tsx file.
Imported Header component into App component.
Deleted header div that was representing Header component. 
Added logo img to Header component
Styled Header component:	
fixed div to size, added image to div and sized it, 		
Added box shadow.

8
Created instructions directory to hold Instructions component files: tsx, scss.
Created Instructions.tsx and scss files.
Inside Instructions.tsc created Instructions components using arrow function. 
Imported global scss variables into Insructions.scss and linked Instructions.scss to Instructions.tsx.
Imported Instructions component into App component.
Replaced the gettingStarted div with the Instructions Component. 
Added getting started button to Instructions component. This button will expand this component and show the necessary requirements to getting started.
Styled getting started button.
Added google fonts link to html doc.


9
Created a typescriptHelperFiles directory to store tsx helper files.
Created a models.tsx file to store component prop interfaces.
Created instructions props interface in models.tsx.
Imported useState hook to app component.
Used useState hook to init data (areInstructionsDisplayed % setAreInstructionsDisplayed) to help toggle instructions div.
Passed init data as prop to instructions component.

10
Used passed props in Instructions component as a conditional to show a Getting Started button or a list of instructions and requirements to getting started.
Styled both outcomes. 
Added toggle functionality.


11
Added box-shadow style to Instructions component.

Manuelhhinojosa committed:
Created styles branch
Deleted Header component file and its styles sass doc.
Deleted Header directory from src folder.
Deleted button & file input from app component
Deleted Header component request from App doc
Deleted styles rules styling header component from styles.scss doc and stablished styles.scss doc as a global styles doc.
Added App.scss doc and stablished/requested it in App.tsx for styling.
Imported SASS variables to App.scss
Deleted styles.scss import from App.tsx as it was requested and executed already in index.tsx
Removed empty code lines from variables.scss
Now clean canvas to work on styles

Manuelhhinojosa committed:
Created main app container in app.tsx
Started styling main app container;
centered it, added a border with box shadow and added margins.

Manuelhhinojosa committed:
Styled body scroll bar.

Manuelhhinojosa committed:
changed main container's box shadow color and height

Manuelhhinojosa committed:
Added 5 divs to App components. Each represents a component that will replace div shortly.

Manuelhhinojosa committed:
Styled temporary divs in App components representing the spaces the components will take.
Added responsiveness at 768px. For app component expanding it from 65vw to 90vw.

Manuelhhinojosa committed:
Added 1rem padding to body.
Created components directory inside src folder.
Created Header directory to store Header component.
Inside Header directory created Header component tsx file using arrow function created Header component. Creadete Header.scss file for styling Header component.
Imported ok200 image from media directory  as logo inside Header.tsx file.
Imported Header component into App component.
Deleted header div that was representing Header component.
Added logo img to Header component
Styled Header component:
	fixed div to size, added image to div and sized it, 		added box shadow.

Manuelhhinojosa committed:
Created instructions directory to hold Instructions component files: tsx, scss.
Created Instructions.tsx and scss files.
Inside Instructions.tsc created Instructions components using arrow function.
Imported global scss variables into Insructions.scss and linked Instructions.scss to Instructions.tsx.
Imported Instructions component into App component.
Replaced the gettingStarted div with the Instructions Component.
Added getting started button to Instructions component. This button will expand this component and show the necessary requirements to getting started.
Styled getting started button.
Added google fonts link to html doc.

Manuelhhinojosa committed:
Created a typescriptHelperFiles directory to store tsx helper files.
Created a models.tsx file to store component prop interfaces.
Created instructions props interface in models.tsx.
Imported useState hook to app component.
Used useState hook to init data (areInstructionsDisplayed % setAreInstructionsDisplayed) to help toggle instructions div.
Passed init data as prop to instructions component.

Manuelhhinojosa committed:
Used passed props in Instructions component as a conditional to show a Getting Started button or a list of instructions and requirements to getting started.
Styled both outcomes.
Added toggle functionality.

Manuelhhinojosa committed:
Added box-shadow style to Instructions component.

scottc11 commented on src/components/Header/Header.tsx:
```diff
@@ -1,14 +1,11 @@
-
-export default Header;
\ No newline at end of file
+export const Header: React.FC = () => {
```
For this export syntax, you should tac on `default`, because React components should only ever export one function (the component function). So like this:
```
export default const Header: React.FC = () => {
```

scottc11 commented on src/App.tsx:
```diff
@@ -1,25 +1,33 @@
+import React, { useState } from "react";
+import dfu from "./dfu-util-js/dfu";
+import "./App.scss";
+import { Header } from "./components/Header/Header";
```
As I mentioned in my other comment, once the `Header.tsx` file uses the `export default` syntax, you can then remove the brackets in `{ Header }` and just import it like this:
```
import Header from "./components/Header/Header";
```

Manuelhhinojosa committed:
fixed code based on comments:
export and import statements.

Manuelhhinojosa committed:
Styled instructions container scrollbar and removed border radius in Instructions component

Manuelhhinojosa committed:
Created ProressBar directory. Created ProgessBar tsx file and component
Created ProgessBar.scss and imported into ProgressBar.tsx
Imported variables.scss into ProgressBar.tsx
Deleted temp div and replaced it with ProgressBar component that I imported into App.tsx
Styles progressBar component.

Manuelhhinojosa committed:
ProgessBar at this point lacks the actual progress functionality.
Working at the moment in adding components that will add the progress functionality to the ProgressBar component

Manuelhhinojosa committed:
Created Steps directory. Created Steps tsx file and component
Created Steps.scss and imported into Steps.tsx
Imported variables.scss into Steps.scss
Deleted temp div and replaced it with Steps
component that I imported into App.tsx
Added containers in Steps components. These containers will hold the html elements that compose this component.
Added and styled check input that eventually will connect with ProgressBar component to work dynamic.

Manuelhhinojosa committed:
Added 3 div containers to step one container
Added title for step 1
Added input type text for step 1
Added instructions text for step 1

Manuelhhinojosa committed:
Added and styled button element to step 2 container.
Added 2 div containers to step3 one container. These div will hold the html elements that will help complete step 3.
Added text that will be replaced by actual instructions.
Added input type file element
Styled both elements
Added and styled button element to step 4.

Manuelhhinojosa committed:
Fixed styles bug in progress bar and steps components

Manuelhhinojosa committed:
added style to buttons in steps component

Manuelhhinojosa committed:
Fixed style bug in steps component
Created Message directory. Created Message tsx file and component
Created Message.scss and imported into Message.tsx
Imported variables.scss into Message.scss
Deleted temp div and replaced it with Steps
component that I imported into App.tsx
Added text to Message component and styled it.
Styled text container.
Init data in App component using useState react hook, this data will give functionality to determine if Message components is displayed or not, this is init to true, which means app will load with success message.

Manuelhhinojosa committed:
Fixed progress bar style bug.
Tested deleting App.css and App.css.map to confirm if deleting these type of docs would break the app, It does not.
Deleted all css and css.map documents
